examples/rs422/rs422_main.c

```diff
@@ -0,0 +1,776 @@
+const int offset, cmd_reqest_t *pget_cmd)
+{
+  int ret;
+  cmd_reqest_t put_cmd;
+
+  memset(&put_cmd, 0, sizeof(cmd_reqest_t));
+  strcpy(put_cmd.cmd, msg);
+  put_cmd.len = len;
+  put_cmd.offset = offset;
+  ret = tty_send_cmd(comm_tty, &put_cmd);
+  if (ret < 0)
+    {
+      printf("could not send command on [%s message]\n", msg);
+      return -1;
+    }
+
+  ret = tty_recv_cmd(comm_tty, pget_cmd);
+  if (ret < 0)
+    {
+      printf("could not recv command on [%s message]\n", msg);
+      return -1;
+    }
+
+  return 0;
+}
+
+/**
+ * @brief RS-485 Message Pathing
+ *
+ * @param comm_tty file descriptor of ttyS2
+ * @param msg send message
+ * @param len len - recv buffer
+ * @param offset offset - recv buffer
+ * @param pget_cmd recv buffer
+ * @return int
+ */
+
+static int
+handshake_master_data
+(int comm_tty, const int len, uint8_t *pbuf)
+{
+  int ret;
+  cmd_reqest_t put_cmd;
+  int remain = len;
+  int offset = 0;
+
+  while (remain > 0)
+    {
+      int reqsize = (remain > CONFIG_UART2_RXBUFSIZE)?
+        CONFIG_UART2_RXBUFSIZE : remain;
+
+      memset(&put_cmd, 0, sizeof(cmd_reqest_t));
+      strcpy(put_cmd.cmd, "dat");
+      put_cmd.len = reqsize;
+      put_cmd.offset = offset;
+      ret = tty_send_cmd(comm_tty, &put_cmd);
+      if (ret < 0)
+        {
+          printf("could not send command on [%s message]\n", put_cmd.cmd);
+          return -1;
+        }
+
+      ret = tty_recv_data(comm_tty, pbuf, offset, reqsize);
+      if (ret < 0)
+        {
+          printf("could not recv command on [%s message]\n", put_cmd.cmd);
+          return -1;
+        }
+
+      remain -= reqsize;
+      offset += reqsize;
+    }
+
+  return 0;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int
+main
+(int argc, char *argv[])
+{
+  int ret;
+  int comm_tty = -1;
+  mode_t mode = APP_MODE_LOOPBACK;
+
+  printf("RS-422 <Test>\n");
+  printf("UART2 Buffer Size = %dbyte\n", CONFIG_UART2_RXBUFSIZE);
+  printf("Communicate Data Set Size = %dbyre\n", sizeof(cmd_reqest_t));
+
+  /* init */
+
+  init_pins();
+  gpio_clear(LED0);
+  gpio_clear(LED1);
+  gpio_clear(LED2);
+  gpio_clear(LED3);
+
+  comm_tty = open_and_setup_ttys();
+  if (comm_tty < 0)
+    {
+      printf("could not open (%s)\n", RS422_TTY_NAME);
+      return -1;
+    }
+
+  /* Check Argument */
+
+  if (argc >= 2)
+    {
+      if (strcmp(argv[1], "-l") == 0)
+        {
+          mode = APP_MODE_LOOPBACK;
+          gpio_set(LED2);
+          gpio_set(LED3);
+        }
+      else if (strcmp(argv[1], "-m") == 0)
+        {
+          mode = APP_MODE_MASTER;
+          gpio_set(LED0);
+        }
+      else if (strcmp(argv[1], "-s") == 0)
+        {
+          mode = APP_MODE_SLAVE;
+          gpio_set(LED1);
+        }
+      else
+        {
+          printf("Error : %s is not support argument.\n", argv[1]);
```

Comment:
```suggestion
          printf("Error : %s is not supported.\n", argv[1]);
```
And, it is helpful to display the usage.